public/app/features/alerting/unified/components/receivers/form/fields/OptionField.tsx

```diff
@@ -75,33 +83,34 @@ export const OptionField: FC<Props> = ({
         pathPrefix={optionPath}
         readOnly={readOnly}
         pathIndex={pathPrefix}
+        parentOption={parentOption}
         customValidator={customValidator}
-        isSecure={isSecure}
+        onResetSecureField={onResetSecureField}
+        secureFields={secureFields}
       />
     </Field>
   );
 };
 
-const OptionInput: FC<Props & { id: string; pathIndex?: string; isSecure?: boolean }> = ({
+const OptionInput: FC<Props & { id: string; pathIndex?: string }> = ({
   option,
   invalid,
   id,
   pathPrefix = '',
   pathIndex = '',
   readOnly = false,
   customValidator,
-  isSecure,
+  onResetSecureField,
+  secureFields = {},
+  parentOption,
 }) => {
   const styles = useStyles2(getStyles);
   const { control, register, unregister, getValues, setValue } = useFormContext();
+
   const name = `${pathPrefix}${option.propertyName}`;
+  const nestedKey = parentOption ? `${parentOption.propertyName}.${option.propertyName}` : option.propertyName;
 
-  useEffect(() => {
-    // Remove the value of secure fields so it doesn't show the incorrect value when clearing the field
-    if (isSecure) {
-      setValue(name, null);
-    }
-  }, [isSecure, name, setValue]);
```

Comment:
Why we no longer need that?

Reply:
these changes were done in this [PR](https://github.com/grafana/grafana/commit/690fbe6077065a6b5dbfd58de9ae1ec3278589f3#diff-ec21c6bb94a9c7ef7dfb616da4155f8f48dd79ce4258f1ae05c5c4117f5493e8R99-R106), to handle what we are fixing the current PR. So, we don't need them anymore as we are solving the same problem.